linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java

```diff
@@ -105,10 +105,18 @@ public static void main(String[] arguments)
       RepositorySystem repositorySystem, RepositorySystemSession session, Path projectDirectory) {
     ImmutableMap.Builder<String, String> artifactToVersion = ImmutableMap.builder();
     Iterable<Path> paths = MoreFiles.fileTraverser().breadthFirst(projectDirectory);
+
     for (Path path : paths) {
       if (!path.getFileName().endsWith("pom.xml")) {
         continue;
       }
+
+      ImmutableSet<Path> elements = ImmutableSet.copyOf(path);
+      if (elements.contains(Paths.get("build")) || elements.contains(Paths.get("target"))) {
```

Comment:
Does it matter where in the path this directory occurs? Can it be only at the end? What if a something several levels up has the name build or target?

Reply:
I would say it does not matter.

From my experience, it's a common practice to avoid using "build" or "target" directory to place something to be published.

Reply:
Another logic I have in mind: flags to indicate the existence of "build.gradle" and "pom.xml" files. (It gets complicated as it seeks for accuracy)

Reply:
Perhaps we can talk about this offline, but I don't think this one is answered. As written, this seems buggy to me.